src/test/java/io/pravega/connectors/flink/FlinkPravegaReaderTest.java

```diff
@@ -9,186 +9,331 @@
-import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment;
-import org.apache.flink.streaming.util.StreamingMultipleProgramsTestBase;
-import org.apache.flink.streaming.util.serialization.AbstractDeserializationSchema;
-
-import org.junit.AfterClass;
-import org.junit.BeforeClass;
-import org.junit.Rule;
+import io.pravega.client.ClientConfig;
+import io.pravega.client.admin.ReaderGroupManager;
+import io.pravega.client.segment.impl.Segment;
+import io.pravega.client.stream.EventPointer;
+import io.pravega.client.stream.EventRead;
+import io.pravega.client.stream.EventStreamReader;
+import io.pravega.client.stream.Position;
+import io.pravega.client.stream.ReaderGroup;
+import io.pravega.client.stream.ReaderGroupConfig;
+import io.pravega.client.stream.Sequence;
+import io.pravega.client.stream.Stream;
+import io.pravega.client.stream.StreamCut;
+import io.pravega.client.stream.impl.EventReadImpl;
+import io.pravega.client.stream.impl.StreamCutImpl;
+import io.pravega.connectors.flink.utils.IntegerDeserializationSchema;
+import io.pravega.connectors.flink.utils.StreamSourceOperatorTestHarness;
+import org.apache.flink.api.common.serialization.DeserializationSchema;
+import org.apache.flink.api.common.time.Time;
+import org.apache.flink.api.java.utils.ParameterTool;
+import org.apache.flink.streaming.api.TimeCharacteristic;
+import org.apache.flink.streaming.api.functions.source.SourceFunction;
+import org.apache.flink.streaming.runtime.streamrecord.StreamRecord;
+import org.apache.flink.streaming.util.TestHarnessUtil;
 import org.junit.Test;
-import org.junit.rules.Timeout;
 
-import java.io.IOException;
-import java.nio.ByteBuffer;
 import java.util.Collections;
-import java.util.concurrent.TimeUnit;
+import java.util.Properties;
+import java.util.Queue;
+import java.util.concurrent.ConcurrentLinkedQueue;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.fail;
+import static org.mockito.Matchers.anyLong;
+import static org.mockito.Mockito.doNothing;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
 
 /**
- * Automated tests for {@link FlinkPravegaReader}.
+ * Unit tests for {@link FlinkPravegaReader} and its builder.
  */
-@Slf4j
-public class FlinkPravegaReaderTest extends StreamingMultipleProgramsTestBase {
-
-    // Number of events to produce into the test stream.
-    private static final int NUM_STREAM_ELEMENTS = 10000;
-
-    // Setup utility.
-    private static final SetupUtils SETUP_UTILS = new SetupUtils();
-
-    //Ensure each test completes within 120 seconds.
-    @Rule
-    public final Timeout globalTimeout = new Timeout(120, TimeUnit.SECONDS);
```

Comment:
It looks like these test cases have been left timeout-less.

Reply:
The unit tests don't wait for anything in an unbounded way that would necessitate a timeout.   I'll review whether any of integration tests (which I agree should use a timeout) are lacking a timeout.